gradle-jdks-settings/src/test/groovy/com/palantir/gradle/jdks/settings/ToolchainJdksSettingsPluginTest.groovy

```diff
@@ -0,0 +1,102 @@
+               
+                daemonTarget = '17'
+            }
+        """.replace("JDK_17_DISTRO", GradleJdkTestUtils.quoted(GradleJdkTestUtils.JDK_17.getLeft()))
+                .replace("JDK_17_VERSION", GradleJdkTestUtils.quoted(GradleJdkTestUtils.JDK_17.getRight()))
+                .stripIndent(true)
+
+        when:
+        file('gradle.properties') << 'palantir.jdk.setup.enabled=true'
+        runTasksSuccessfully("generateGradleJdkConfigs")
+
+        then: 'only gradle configuration files are generated, no jdks are installed'
+        String os = CurrentOs.get().uiName()
+        String arch = CurrentArch.get().uiName()
+        Path jdk17LocalPath = projectDir.toPath().resolve("gradle/jdks/17/${os}/${arch}/local-path")
+        String originalJdk17LocalPath = jdk17LocalPath.text.trim()
+        Path originalJdkPath = Path.of(System.getProperty("user.home")).resolve(".gradle/gradle-jdks")
+                .resolve(originalJdk17LocalPath).toAbsolutePath()
+        !Files.exists(originalJdkPath)
+
+        when: 'trigger a task'
+        jdk17LocalPath.text = "amazon-corretto-${gradleVersion}-test1\n"
+        ExecutionResult executionResult = runTasksSuccessfully("javaToolchains")
+
+        then: 'the jdks are installed by the settings plugin'
+        executionResult.standardError.contains("Gradle JDK setup is enabled (palantir.jdk.setup.enabled is true)" +
+                " but some jdks were not installed")
+        executionResult.standardOutput.contains("Auto-detection:     Disabled")
+        executionResult.standardOutput.contains("Auto-download:      Disabled")
+        executionResult.standardOutput.contains("JDK ${GradleJdkTestUtils.SIMPLIFIED_JDK_17_VERSION}")
+        Path expectedJdkPath = USER_HOME_PATH.resolve(".gradle/gradle-jdks")
+                .resolve("amazon-corretto-${gradleVersion}-test1").toAbsolutePath()
+        Files.exists(expectedJdkPath)
+
+        when: 'if the jdk configured path is changed'
+        jdk17LocalPath.text = "amazon-corretto-${gradleVersion}-test2\n"
+        ExecutionResult resultAfterJdkChange = runTasksSuccessfully("javaToolchains")
+
+        then:
+        resultAfterJdkChange.standardError.contains("Gradle JDK setup is enabled (palantir.jdk.setup.enabled is true)" +
+                " but some jdks were not installed")
+        Path newInstalledJdkPath = USER_HOME_PATH.resolve(".gradle/gradle-jdks")
+                .resolve("amazon-corretto-${gradleVersion}-test2").toAbsolutePath()
+        Files.exists(newInstalledJdkPath)
+
+        cleanup:
+        Files.walk(expectedJdkPath)
+                .sorted(Comparator.reverseOrder())
+                .forEach(Files::delete)
+
+        Files.walk(newInstalledJdkPath)
+                .sorted(Comparator.reverseOrder())
+                .forEach(Files::delete)
+
+        where:
+        gradleVersionNumber << [GradleJdkTestUtils.GRADLE_7_6_VERSION, GradleJdkTestUtils.GRADLE_8_5_VERSION, GradleJdkTestUtils.GRADLE_8_8_VERSION]
```

Comment:
why is this different than the GRADLE_TEST_VERSIONS you put in other tests?

Reply:
I wanted to explicitly test other gradle versions because we are using [reflection in the settings plugin](https://github.com/palantir/gradle-jdks/blob/develop/gradle-jdks-settings/src/main/java/com/palantir/gradle/jdks/settings/ToolchainJdksSettingsPlugin.java#L86-L104), so I want to make sure nothing gets broken in between gradle versions. 

pushed a [commit](https://github.com/palantir/gradle-jdks/pull/382/commits/142fb57dfe05f52c42c4105b8d4a17d446df64a4#diff-47297190c845738f2860d17927cc1cf986df22933651ea980e7345fdc840f6ecR99) that adds a comment as to why these are the gradle versions I am testing with